zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/elasticsearch/ElasticsearchEngineClient.java

```diff
@@ -178,6 +184,34 @@ public void putIndexLifeCyclePolicy(final String policyName, final String deleti
     }
   }
 
+  @Override
+  public boolean importersCompleted(final int partitionId, final String indexPrefix) {
+    final var importPositionSearchRequest = importPositionDocuments(partitionId, indexPrefix);
+
+    try {
+      final var docs = client.search(importPositionSearchRequest, ImportPositionEntity.class);
+      return allImportersCompleted(
+          docs.hits().hits().stream().map(Hit::source).toList(), ImportValueType.values().length);
+    } catch (final IOException e) {
+      final var errMsg =
+          String.format(
+              "Failed to search documents in the import position index for partition [%s]",
+              partitionId);
+      LOG.error(errMsg, e);
+      throw new ElasticsearchExporterException(errMsg, e);
+    }
+  }
+
+  private SearchRequest importPositionDocuments(final int partitionId, final String indexPrefix) {
+    final var importPositionIndexName =
+        new ImportPositionIndex(indexPrefix, true).getFullQualifiedName();
+    return new SearchRequest.Builder()
+        .index(importPositionIndexName)
+        .size(ImportValueType.values().length)
+        .query(q -> q.match(m -> m.field(ImportPositionIndex.PARTITION_ID).query(partitionId)))
```

Comment:
🔧 ❓ We could already match here the completed equals true right?

Reply:
I don't think so, if we also match the completed field then we cannot differentiate between a genuinely empty index and an index where all the documents have completed=false, we could add a query for the amount of documents to differentiate but at that point that's an unnecessary extra query when we could just query all the documents with one request. There will be at most a dozen documents anyway.

Reply:
I refactored the importers completed logic to also match the completed field